azure-pipelines.yml

```diff
@@ -9,8 +9,25 @@ trigger:
 pool:
   vmImage: ubuntu-latest
 
+variables:
+  - group: docker-settings
+
 steps:
   - script: docker-compose -f docker-compose.test.yml build
-    displayName: Building Docker container
+    displayName: Building Docker Image for tests
   - script: docker-compose -f docker-compose.test.yml run --rm web bundle exec rake
     displayName: Running rake default
+
+  - script: docker login -u $(dockerId) -p $pass
+    env:
+      pass: $(dockerPassword)
+    displayName: Docker login
+    condition: eq(variables['Build.SourceBranch'], 'refs/heads/master')
+  - script: docker build --file=Dockerfile --tag=$(dockerRegistry)/$(imageName):latest --target=web .
```

Comment:
Can we do a `git reset --hard && git clean -xdf` before this step to ensure nothing from testing leaks in? I can imagine us adding volume linking to the test docker compose config, and getting some leakage.